test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/Common/Queue/AzureStorageQueueManager.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Common
+{
+	// Implementation of IQueueManager responsible for management of Azure Queue Resource.
+	public class AzureStorageQueueManager : IQueueManager<QueueRequest, QueueResponse>
+	{
+		private readonly string _connectionString;
+		private static readonly AzureStorageQueueManager _instance = new ();
+		private readonly ConcurrentDictionary<string, QueueClient> _queueClientStore;
+		private readonly ILogger _logger = TestLogger.GetTestLogger();
+
+		public static AzureStorageQueueManager GetInstance()
+		{
+			return _instance;
+		}
+
+		private AzureStorageQueueManager()
+		{
+			_connectionString = Environment.GetEnvironmentVariable(Constants.AZURE_WEBJOBS_STORAGE);
+			_queueClientStore = new ConcurrentDictionary<string, QueueClient>();
+		}
+
+		public async Task ClearAsync(string queueName)
+		{
+			QueueClient queueClient = new QueueClient(_connectionString, queueName);
+			await queueClient.CreateIfNotExistsAsync();
+			await queueClient.ClearMessagesAsync();
+			_queueClientStore.GetOrAdd(queueName, queueClient);
+			_logger.LogInformation($"Clearing the queue: {queueName}");
+		}
+
+		public Task CreateAsync(string queueName)
```

Comment:
I might be missing here , I haven't seen the implementation yet, But shouldnt this function take care of creating the clients in the first hand and others should just assert ( validate) that it exist in the cache ? Why should read/clear have create clients?

Reply:
In our current implementation we do not assume that a create method is called before the clear command as we are persisting the azure queue even after the test cases are completed for debugging purposes. Have updated this to a cleaner implementation to make the idea more clear.